### PR TITLE
SILGen: Fix for 'try?' leaking the error value

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please fix

File: test/Interpreter/errors.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var count = 0
+
+enum Markings { case Tuxedo, Tortoise }
+
+class Cat : Equatable {
```
Comment (gribozavr) [inline test/Interpreter/errors.swift]: I like cats, but StdlibUnittest provides `LifetimeTracked` type for leak checking.  And the harness automatically checks that `LifetimeTracked.instances == 0` at the end of every test.

**Comment by slavapestov** [comment]
@swift-ci Please test

File: test/Interpreter/errors.swift (hunk 1 above)
Comment (slavapestov) [inline test/Interpreter/errors.swift]: Thanks! And I'm glad we both agree on the cat part too.

Comment (dduan) [inline test/Interpreter/errors.swift]: Tortoise = naughty.

**Comment by natecook1000** [comment]
> @swift-ci Please fix

If only this worked...

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by gparker42** [comment]
> > @swift-ci Please fix

> If only this worked...

Let me try something…
@swift-ci Please implement "@swift-ci Please fix"

**Comment by dduan** [comment]
I'm guessing this won't make it into Xcode 7.3 GM…?

**Comment by dduan** [comment]
May I suggest this one being cherry-picked for the (potential) 2.2.1 release? It'll have big impact in our production code.

Thanks for your consideration 🙇.

@tkremenek @DougGregor

**Pull request closed without merging**